src/c++/perf_analyzer/load_manager.cc

```diff
@@ -521,6 +527,7 @@ LoadManager::PrepareSharedMemoryInfer(InferContext* ctx)
         &infer_input, backend_->Kind(), input.first, shape,
         input.second.datatype_));
     ctx->inputs_.push_back(infer_input);
+    ctx->valid_inputs_.push_back(infer_input);
```

Comment:
Why are we adding all the inputs in valid_inputs_ for shared_memory?
Please look at the `InitSharedMemory` function above. It creates shared memory region and copies the data over to the shared memory. 
I propose if the data is not found then we must not create a shared memory region for the same and skip it in valid_inputs_.
https://github.com/triton-inference-server/client/blob/main/src/c%2B%2B/perf_analyzer/load_manager.cc#L356

Reply:
I consulted with the requestor of this optional inputs feature, and they are not using the shared memory feature of PA, so it seemed useful to postpone implementing the optional inputs feature for shared memory. There is a separate ticket for that.

The reason `valid_inputs_` is added here in the shared memory function is to preserve the original non-optional-inputs functionality of the shared memory feature. Without this, the shared memory mode would be passing an empty `valid_inputs_` vector to the inference context in `ConcurrencyManager`/`RequestRateManager`.

Reply:
Can you add a comment here about the current and expected behavior as a FIXME at the top?

Reply:
Added 👍